src/include/simeng/arch/riscv/Architecture.hh

```diff
@@ -14,15 +14,21 @@ namespace simeng {
 namespace arch {
 namespace riscv {
 
+struct archConstants {
+  uint8_t alignMask;
+  uint8_t bytesLimit; /* Minimum bytes the decoder needs to process */
+  uint8_t regWidth;   /* Register width in bytes */
+};
+
```

Comment:
This is lay over from Huawei implementation which included 32 bit support and therefore wider use of this struct. I kept it as it was neat and coagulated variables with similar usages. It may seem slightly odd now but if/when 32 bit support is added this is a good way of doing things in my opinion. Breaking up into normal fields would be fine but you lose the grouping. Thoughts?

Reply:
Is 32-bit support something that you have planned in the near future? If yes, then I can see the point of this struct's contents and the one in ArchInfo. If it isn't in our plans anytime soon however, I feel as though there removal would make the current code a bit cleaner and easier to follow (not that it is hard to follow right now, or untidy, just my opinion on it as it introduces new variables where they aren't currently required).

In either case, I'm not 100% sure on having these values inside a struct. They're internal to the Architecture class and describe its current state. Having them as member variables would align with the current code base better in my view